packages/headless/src/controllers/insight/search-box/headless-insight-search-box.ts

```diff
@@ -40,9 +48,41 @@ export function buildSearchBox(
   engine: InsightEngine,
   props: SearchBoxProps = {}
 ): SearchBox {
-  return buildCoreSearchBox(engine, {
+  const {dispatch} = engine;
+  const searchBox = buildCoreSearchBox(engine, {
```

Comment:
Instead of reimplementing part of the search box, could the analytics action be an option on the `headless-core-search-box` ?

Reply:
perhaps the `submit` function on the core searchbox could take the analytics as a parameter, and both the search & insight search boxes would "re-implement" the function without parameters by calling the core search box's with the correct parameter.

Reply:
Thank you for the suggestions, I went with the second one! let me what you think 👌